app/models/cms/email_template.rb

```diff
@@ -82,7 +82,7 @@ def mime_type
 
   def headers=(val)
     @headers = nil
-    self[:options] = val
+    self[:options] = val.respond_to?(:to_unsafe_h) ? val.to_unsafe_h : val
```

Comment:
Although it isn't exactly the same, another option could be (depending on our case, i dunno if this is good or not):
```ruby
self[:options] = val.try(:to_unsafe_h) || val
```
which is actually equivalent to:
```ruby
self[:options] = val.respond_to?(:to_unsafe_h) ? (val.to_unsafe_h || val) : val
```

Reply:
I think I prefer the `respond_to?` version
Normally this `val` is an `ActionController::Parameters`
A good way to do it is:


```suggestion
self[:options] = val.to_unsafe_h
rescue NoMethodError
  ActiveSupport::Deprecation.warn('Passing a value that does not respond to `to_unsafe_h` to #headers= ')
 val
```

Though I do not want to do that, we might have tons of them ...

Reply:
good then :+1: :wink: